rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -287,18 +309,21 @@ def default_call_order(self, skip_etherscan: bool = False) -> list[WeightedNode]
         ===> Runs: 66, 82, 72, 58, 72 seconds
         ---> Average: 70 seconds
         """
-        open_nodes = self.database.get_rpc_nodes(blockchain=self.blockchain, only_active=True)
+        open_nodes = self.database.get_rpc_nodes(
+            blockchain=self.blockchain,
+            only_active=True,
+            include_zero_weighted=False,
+        )
         if skip_etherscan:
             selection = [wnode for wnode in open_nodes if wnode.node_info.name != self.etherscan_node_name and wnode.node_info.owned is False]  # noqa: E501
         else:
             selection = [wnode for wnode in open_nodes if wnode.node_info.owned is False]
 
-        ordered_list = []
-        while len(selection) != 0:
-            weights = [float(entry.weight) for entry in selection]
-            node = random.choices(selection, weights, k=1)
-            ordered_list.append(node[0])
-            selection.remove(node[0])
```

Comment:
here I wanted to avoid this append + remove. Instead use a list cvomprehension and avoid the while loop